cli/ops/runtime_compiler.rs

```diff
@@ -71,16 +77,52 @@ struct TranspileArgs {
   options: Option<String>,
 }
 
+#[derive(Debug, Serialize)]
+struct RuntimeTranspileEmit {
+  source: String,
+  map: Option<String>,
+}
+
 async fn op_transpile(
   state: Rc<RefCell<OpState>>,
   args: Value,
   _data: BufVec,
 ) -> Result<Value, AnyError> {
   super::check_unstable2(&state, "Deno.transpile");
   let args: TranspileArgs = serde_json::from_value(args)?;
-  let cli_state = super::global_state2(&state);
-  let program_state = cli_state.clone();
-  let result =
-    runtime_transpile(program_state, &args.sources, &args.options).await?;
+
+  let mut compiler_options = tsc_config::TsConfig::new(json!({
+    "esModuleInterop": true,
+    "module": "esnext",
+    "sourceMap": true,
+    "scriptComments": true,
+    "target": "esnext",
+  }));
+
+  let user_options = if let Some(options) = args.options {
+    tsc_config::parse_raw_config(&options)?
+  } else {
+    json!({})
+  };
+  compiler_options.merge(&user_options);
+
+  let emit_options: ast::EmitOptions = compiler_options.into();
+  let mut emit_map = HashMap::new();
+
+  for (specifier, source) in args.sources {
+    let media_type = MediaType::from(&PathBuf::from(&specifier));
+    let module_specifier = ModuleSpecifier::resolve_url(&specifier)?;
```

Comment:
This is a big change and I think it will cause tests to fail - currently `Deno.traspileOnly()` doesn't require valid module specifier. I had to resolve the specifier here, because it's required by `ast::parse`. We'll have to discuss this one.

Reply:
This problem can be circumvented by rolling back `ast::parse` to take specifier as `&str` instead of `&ModuleSpecifier`

Reply:
I tentatively applied this fix

Reply:
Yeah, I guess `ast::parse` doesn't need to be a `&ModuleSpecifier`... I just wanted to use `ModuleSpecifier` everywhere and just convert at the edge, so I think `ast::parse` going back to `&str` would be fine.  I guess it shouldn't really have "Deno" specific opinions where avoidable.